pandas/core/interchange/from_dataframe.py

```diff
@@ -378,16 +386,22 @@ def datetime_column_to_ndarray(col: Column) -> tuple[np.ndarray | pd.Series, Any
     buffers = col.get_buffers()
 
     _, _, format_str, _ = col.dtype
-    dbuf, dtype = buffers["data"]
-    # Consider dtype being `uint` to get number of units passed since the 01.01.1970
-    data = buffer_to_ndarray(
-        dbuf,
-        (
+    dbuf, data_dtype = buffers["data"]
+
+    if data_dtype[0] == DtypeKind.DATETIME:
+        # temporary workaround to keep backwards compatibility due to
+        # https://github.com/pandas-dev/pandas/issues/54781
+        # Consider dtype being `uint` to get number of units passed since the 01.01.1970
+        data_dtype = (
             DtypeKind.UINT,
```

Comment:
Shouldn't this be an INT? Timestamps are backed by 64 bit signed integers in arrow

https://github.com/apache/arrow/blob/772a01c080ad57eb11e9323f5347472b769d45de/format/Schema.fbs#L264

Reply:
Yes, AFAIK that should be signed INT